source/base/hdf5.cc

```diff
@@ -386,7 +386,7 @@ namespace HDF5
     T                            value;
     hid_t                        attr;
     herr_t                       ret;
-
+    (void)ret;
```

Comment:
This looks out of place. In particular, it's *above* the writes to the variable, so a compiler might reasonably still complain that there are dead writes to the variable without any corresponding reads. Can you move this right above the assertion?